also build fail likely real
```
======================================================================
ERROR: test_user_registration (corehq.apps.app_manager.tests.test_views.TestViews)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/app_manager/tests/test_views.py", line 164, in test_user_registration
    'app_id': self.app.id,
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/app_manager/tests/test_views.py", line 111, in _test_status_codes
    response = self.client.get(reverse(name, kwargs=kwargs), follow=False)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/test/client.py", line 470, in get
    **extra)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/test/client.py", line 286, in get
    return self.generic('GET', path, secure=secure, **r)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/test/client.py", line 358, in generic
    return self.request(**r)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/test/client.py", line 440, in request
    six.reraise(*exc_info)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 111, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/domain/decorators.py", line 81, in _inner
    return view_func(req, domain_name, *args, **kwargs)
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/users/decorators.py", line 21, in _inner
    return view_func(request, domain, *args, **kwargs)
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/app_manager/views/form_designer.py", line 183, in user_registration_source
    return form_designer(request, domain, app_id, is_user_registration=True)
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/domain/decorators.py", line 81, in _inner
    return view_func(req, domain_name, *args, **kwargs)
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/users/decorators.py", line 21, in _inner
    return view_func(request, domain, *args, **kwargs)
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/app_manager/views/form_designer.py", line 80, in form_designer
    if getattr(module, 'has_schedule', False):
UnboundLocalError: local variable 'module' referenced before assignment
```

--- a/corehq/apps/app_manager/views/form_designer.py
+++ b/corehq/apps/app_manager/views/form_designer.py
@@ -13,6 +13,12 @@ from corehq.apps.app_manager.views.apps import get_apps_base_context
 from corehq.apps.app_manager.views.utils import back_to_main, bail
 from corehq import toggles, privileges
 from corehq.apps.accounting.utils import domain_has_privilege
+from corehq.apps.app_manager.const import (
+    SCHEDULE_CURRENT_VISIT_NUMBER,
+    SCHEDULE_NEXT_DUE,
+    SCHEDULE_UNSCHEDULED_VISIT,
+    SCHEDULE_GLOBAL_NEXT_VISIT_DATE,
+)
 from corehq.apps.app_manager.util import (
     get_casedb_schema,
     get_session_schema,
@@ -34,6 +40,7 @@ logger = logging.getLogger(__name__)
 def form_designer(request, domain, app_id, module_id=None, form_id=None,
                   is_user_registration=False):
     app = get_app(domain, app_id)
+    module = None
 
     if is_user_registration:
         form = app.get_user_registration()
@@ -76,17 +83,22 @@ def form_designer(request, domain, app_id, module_id=None, form_id=None,
         'custom_intents': domain_has_privilege(domain, privileges.CUSTOM_INTENTS),
     })
 
+    has_schedule = (
+            getattr(module, 'has_schedule', False) and
+            getattr(form, 'schedule', False) and form.schedule.enabled)
     scheduler_data_nodes = []
-    if getattr(module, 'has_schedule', False):
+    if has_schedule:
         scheduler_data_nodes = [
-            "unscheduled_visit",
-            "current_visit_number",
-            "next_visit",
-            "next_visit_date",
-            "next_due",
+            SCHEDULE_CURRENT_VISIT_NUMBER,
+            SCHEDULE_NEXT_DUE,
+            SCHEDULE_UNSCHEDULED_VISIT,
+            SCHEDULE_GLOBAL_NEXT_VISIT_DATE,
         ]
-        scheduler_data_nodes.extend([u"next_{}".format(form_.schedule_form_id)
-            for form_ in module.get_forms() if form_.schedule_form_id])
+        scheduler_data_nodes.extend([
+            u"next_{}".format(f.schedule_form_id)
+            for f in form.get_phase().get_forms
+            if getattr(f, 'schedule', False) and f.schedule.enabled
+        ])
 
 
     context = get_apps_base_context(request, domain, app)

--- a/corehq/apps/app_manager/views/form_designer.py
+++ b/corehq/apps/app_manager/views/form_designer.py
@@ -97,7 +97,7 @@ def form_designer(request, domain, app_id, module_id=None, form_id=None,
         ]
         scheduler_data_nodes.extend([
             u"next_{}".format(f.schedule_form_id)
-            for f in form.get_phase().get_forms
+            for f in form.get_phase().get_forms()
             if getattr(f, 'schedule', False) and f.schedule.enabled
         ])
 

--- a/corehq/apps/app_manager/views/form_designer.py
+++ b/corehq/apps/app_manager/views/form_designer.py
@@ -84,8 +84,9 @@ def form_designer(request, domain, app_id, module_id=None, form_id=None,
     })
 
     has_schedule = (
-            getattr(module, 'has_schedule', False) and
-            getattr(form, 'schedule', False) and form.schedule.enabled)
+        getattr(module, 'has_schedule', False) and
+        getattr(form, 'schedule', False) and form.schedule.enabled
+    )
     scheduler_data_nodes = []
     if has_schedule:
         scheduler_data_nodes = [

